src/puppetlabs/puppetdb/command.clj

```diff
@@ -428,20 +414,27 @@
         (recur (dec n))
```

Comment:
My guess is where you see catch Throwable, the purpose it served was catching AssertionErrors. In other areas I've been splitting that out, my guess is that's the same reason it was originally used here.